### PR TITLE
Add unit tests for Identity Config Parser

### PR DESCRIPTION
**Pull request by pulasthi7**
### Proposed changes in this pull request

Unit tests for Identity Config Parser
Overall line coverage 14% -> 21%

#### General

- [ ] **Is this PR explained thoroughly?** All code changes must be accounted for in the PR description.
- [ ] **Is the PR labeled correctly?**

#### Functionality

- [ ] **Are all requirements met?** Compare implemented functionality with the requirements specification.
- [ ] **Does the UI work as expected?** There should be no Javascript errors in the console; all resources should load. There should be no unexpected errors. Deliberately try to break the feature to find out if there are corner cases that are not handled.

#### Code

- [ ] **Do you fully understand the introduced changes to the code?** If not ask for clarification, it might uncover ways to solve a problem in a more elegant and efficient way.
- [ ] **Does the PR introduce any inefficient database requests?** Use the debug server to check for duplicate requests.
- [ ] **Are all necessary strings marked for translation?** All strings that are exposed to users via the UI must be [marked for translation](https://docs.djangoproject.com/en/1.10/topics/i18n/translation/).

#### Tests

- [ ] **Are there sufficient test cases?** Ensure that all components are tested individually; models, forms, and serializers should be tested in isolation even if a test for a view covers these components.
- [ ] **If this is a bug fix, are tests for the issue in place?**  There must be a test case for the bug to ensure the issue won’t regress. Make sure that the tests break without the new code to fix the issue.
- [ ] **If this is a new feature or a significant change to an existing feature?** has the manual testing spreadsheet been updated with instructions for manual testing?

#### Security

- [ ] **Confirm this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.**
- [ ] **Are all UI and API inputs run through forms or serializers?**
- [ ] **Are all external inputs validated and sanitized appropriately?**
- [ ] **Does all branching logic have a default case?**
- [ ] **Does this solution handle outliers and edge cases gracefully?**
- [ ] **Are all external communications secured and restricted to SSL?**

#### Documentation

- [ ] **Are changes to the UI documented in the platform docs?** If this PR introduces new platform site functionality or changes existing ones, the changes should be documented.
- [ ] **Are changes to the API documented in the API docs?** If this PR introduces new API functionality or changes existing ones, the changes must be documented.
- [ ] **Are reusable components documented?** If this PR introduces components that are relevant to other developers (for instance a mixin for a view or a generic form) they should be documented in the Wiki.

### CODE REVIEW

File: components/identity-core/org.wso2.carbon.identity.core/src/test/java/org/wso2/carbon/identity/core/util/IdentityConfigParserTest.java [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+    @Test(dependsOnMethods = "testGetInstance")
+    public void testGetConfigElementNegative() throws Exception {
+        assertNull(IdentityConfigParser.getInstance().getConfigElement("NoneExisting"),"Should return null for none" +
+                " existing elements");
+    }
+
+    @Test(dependsOnMethods = "testGetInstance")
+    public void testGetQNameWithIdentityNS() throws Exception {
+        QName element = IdentityConfigParser.getInstance().getQNameWithIdentityNS("TestElement");
```
Comment (omindu) [inline components/identity-core/org.wso2.carbon.identity.core/src/test/java/org/wso2/carbon/identity/core/util/IdentityConfigParserTest.java]: Shouldn't the `TestElement` be in the `identity.xml` ?

File: components/identity-core/org.wso2.carbon.identity.core/src/test/java/org/wso2/carbon/identity/core/util/IdentityConfigParserTest.java [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+        System.clearProperty(ServerConstants.CARBON_HOME);
+    }
+
+    @Test
+    public void testGetInstance() throws Exception {
+        String identityXmlPath = Paths.get(System.getProperty("user.dir"), "src", "test", "resources",
+                "identity.xml").toString();
+        System.setProperty(ServerConstants.CARBON_HOME, ".");
+        IdentityConfigParser.getInstance(identityXmlPath);
+        assertTrue(MapUtils.isNotEmpty(IdentityConfigParser.getInstance().getConfiguration()),"Configuration should " +
```
Comment (omindu) [inline components/identity-core/org.wso2.carbon.identity.core/src/test/java/org/wso2/carbon/identity/core/util/IdentityConfigParserTest.java]: Missing space after `,`. Check other places as well.

File: components/identity-core/org.wso2.carbon.identity.core/src/test/java/org/wso2/carbon/identity/core/util/IdentityConfigParserTest.java (hunk 1 above)
Comment (pulasthi7) [reply components/identity-core/org.wso2.carbon.identity.core/src/test/java/org/wso2/carbon/identity/core/util/IdentityConfigParserTest.java]: No

File: components/identity-core/org.wso2.carbon.identity.core/src/test/java/org/wso2/carbon/identity/core/util/IdentityConfigParserTest.java [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+    public void testGetIdentityCookieConfigurationHolder() throws Exception {
+        assertTrue(MapUtils.isNotEmpty(IdentityConfigParser.getIdentityCookieConfigurationHolder()), "Cookie config " +
+                "holder should not be null/empty");
+    }
+
+    @DataProvider(name = "ConfigElementValues")
+    public Object[][] getConfigElement() {
+        return new Object[][]{
+                {"ClockSkew", "300"},
+                {"EnableAskPasswordAdminUI", "false"},
```
Comment (dewniMW) [inline components/identity-core/org.wso2.carbon.identity.core/src/test/java/org/wso2/carbon/identity/core/util/IdentityConfigParserTest.java]: Isn't the comma at the end unneccesary?

File: components/identity-core/org.wso2.carbon.identity.core/src/test/resources/identity.xml [hunk 4]
```diff
@@ -0,0 +1,381 @@
...
+        <OpenIDUserPattern>${carbon.protocol}://${carbon.host}:${carbon.management.port}/openid</OpenIDUserPattern>
+        <OpenIDLoginUrl>${carbon.protocol}://${carbon.host}:${carbon.management.port}/authenticationendpoint/openid_login.do</OpenIDLoginUrl>
+        <!-- If the users must be prompted for approval -->
+        <OpenIDSkipUserConsent>false</OpenIDSkipUserConsent>
+        <!-- Expiry time of the OpenID RememberMe token in minutes -->
+        <OpenIDRememberMeExpiry>7200</OpenIDRememberMeExpiry>
+        <!-- To enable or disable openid dumb mode -->
+        <DisableOpenIDDumbMode>false</DisableOpenIDDumbMode>
+
+	-->
```
Comment (ashensw) [inline components/identity-core/org.wso2.carbon.identity.core/src/test/resources/identity.xml]: remove unnecessary.

> -->

File: components/identity-core/org.wso2.carbon.identity.core/src/test/resources/identity.xml [hunk 5]
```diff
@@ -0,0 +1,381 @@
...
+            <ClaimsRetrieverImplClass>org.wso2.carbon.identity.oauth2.authcontext.DefaultClaimsRetriever</ClaimsRetrieverImplClass>
+            <ConsumerDialectURI>http://wso2.org/claims</ConsumerDialectURI>
+            <SignatureAlgorithm>SHA256withRSA</SignatureAlgorithm>
+            <AuthorizationContextTTL>15</AuthorizationContextTTL>
+        </AuthorizationContextTokenGeneration>
+
+        <SAML2Grant>
+            <UserType>LOCAL</UserType>
+            <!--UserType>FEDERATED</UserType-->
+            <!--UserType>LEGACY</UserType-->
```
Comment (ashensw) [inline components/identity-core/org.wso2.carbon.identity.core/src/test/resources/identity.xml]: Do we need to have these commented configs?

File: components/identity-core/org.wso2.carbon.identity.core/src/test/resources/identity.xml [hunk 6]
```diff
@@ -0,0 +1,381 @@
...
+        </CacheManager>
+    </CacheConfig>
+
+    <Cookies>
+        <Cookie name="commonAuthId" domain="localhost" httpOnly="true" secure="true" />
+        <Cookie name="dummy" domain="localhost" httpOnly="true" secure="true" path="/" comment="comment" />
+        <Cookie name="dummy"  path="/" comment="comment" version="1" maxAge="100"/>
+    </Cookies>
+
+
```
Comment (dewniMW) [inline components/identity-core/org.wso2.carbon.identity.core/src/test/resources/identity.xml]: unnecessary new line

**Pull request merged**